models/status.go

```diff
@@ -170,6 +172,59 @@ func GetLatestCommitStatus(repo *Repository, sha string, page int) ([]*CommitSta
 	return statuses, x.In("id", ids).Find(&statuses)
 }
 
+// GetLatestCommitStatuses returns all statuses with given repoIDs and shas
+func GetLatestCommitStatuses(repoIDs []int64, shas []string) ([][]*CommitStatus, error) {
+	if len(repoIDs) != len(shas) {
+		return nil, errors.New("parameter repoIDs should have the same size of shas")
+	}
+
+	var results = make([]struct {
+		ID     int64
+		RepoID int64
+	}, 0, 10*len(repoIDs))
+
+	var cond = builder.NewCond()
+	for i := 0; i < len(repoIDs); i++ {
+		cond = cond.Or(builder.Eq{
+			"repo_id": repoIDs[i],
+			"sha":     shas[i],
+		})
+	}
+
+	err := x.Table(&CommitStatus{}).
+		Where(cond).
+		Select("max( id ) as id, repo_id").
+		GroupBy("repo_id, sha, context").OrderBy("max( id ) desc").Find(&results)
```

Comment:
Is not this select everything you need to get latest commit for each repo commit (I was just writing about adding columns to "group by" when you updated it)? You can use this as subselect (just remove `repo_id` from select) for mainy "where" clause.